magister-calendar.js

```diff
@@ -440,8 +446,21 @@ function parseAppointments(appointments, currentcourse) {
 
       // Check if the location is still the same.
       if (cache.location !== appointment.location) {
-        tools.log("notice", appointment.id + " Location has changed.");
-        tools.sendPushMessage(CONFIG.pushover, "location", appointment, cache.location);
+
+        /* Loop through the locations that mean cancelled. */
+        for (var a = 0; a < CONFIG.location_means_cancelled.length; a++) {
+          // If the location means that the appointment was cancelled we cancel it.
+          if (appointment.location == CONFIG.location_means_cancelled[a]) {
+            tools.log("notice", appointment.id + " Location has changed. Appointment was cancelled.");
+            // Don't send a push notification. This will be done when canceling. (line: 556:5)
+            var done = true;
+          }
+        }
+        // If the appointment has not been caught by the loop above we notify the changed location.
+        if(!done) {
+          tools.log("notice", appointment.id + " Location has changed.");
+          tools.sendPushMessage(CONFIG.pushover, "location", appointment, cache.location);
+        }
```

Comment:
Why do you not `break` this loop if a match was found? Also, please use `===` over `==`.